src/utils/sources-tree/addToTree.js

```diff
@@ -12,14 +14,118 @@ import { getURL, getFilenameFromPath } from "./getURL";
 import type { Node } from "./types";
 import type { SourceRecord } from "../../reducers/types";
 
-function createNodeInTree(part: string, path: string, tree: Node) {
+function createNodeInTree(
+  part: string,
+  path: string,
+  tree: Node,
+  index: number
+) {
   const node = createNode(part, path, []);
+
   // we are modifying the tree
-  tree.contents = [...tree.contents, node];
+  const contents = tree.contents.slice(0);
+  contents.splice(index, 0, node);
+  tree.contents = contents;
+
   return node;
 }
 
 /*
+ * Gets domain from url (without www prefix)
+ */
+function getDomain(url: string): ?string {
+  const { host } = parse(url);
+  if (!host) {
+    return null;
+  }
+  return host.indexOf("www.") === 0 ? host.substr(4) : host;
+}
+
+/*
+ * Checks if node name matches debugger host/domain.
+ */
+function isExactDomainMatch(part: string, debuggeeHost: string): boolean {
+  return part.indexOf("www.") === 0
+    ? part.substr(4) === debuggeeHost
+    : part === debuggeeHost;
+}
+
+type FindNodeInContentsMatcher = any => number;
+/*
+ * Performs a binary search to insert a node into contents. Returns positive
+ * number, index of the found child, or negative number, which can be used
+ * to calculate a position where a new node can be inserted (`-index - 1`).
```

Comment:
so, if an index is negative, it is a new node? and if it is positive it is an existing node? that is difficult to understand. why not use a tuple? { index: <number>, newNode: <boolean> }